stdlib/public/core/StringLegacy.swift

```diff
@@ -42,12 +36,32 @@ extension String {
   ///     let zeroes = String("0" as UnicodeScalar, count: 10)
   ///     print(zeroes)
   ///     // Prints "0000000000"
+
+  @available(*, unavailable, message: "Replaced by init(repeating: String, count: Int)")
   public init(repeating repeatedValue: UnicodeScalar, count: Int) {
-    self = String._fromWellFormedCodeUnitSequence(
-      UTF32.self,
-      input: repeatElement(repeatedValue.value, count: count))
+    Builtin.unreachable()
   }
-  
+
+  /// Creates a string representing the given string repeated the
+  /// specified number of times.
+  ///
+  /// For example, use this initializer to create a string with ten `"00"`
+  /// strings in a row.
+  ///
+  ///     let zeroes = String("00", count: 10)
+  ///     print(zeroes)
+  ///     // Prints "00000000000000000000"
+  public init(repeating repeatedValue: String, count: Int) {
```

Comment:
You could consider adding a special-case path for a single ASCII character or one UnicodeScalar.